edenai_apis/apis/microsoft/microsoft_ocr_api.py

```
@@ -364,142 +455,133 @@ def ocr__identity_parser(
-                raise ProviderException(str(provider_call_exception))
+                elif file_url:
+                    poller = await document_analysis_client.begin_analyze_document_from_url(
+                        "prebuilt-idDocument", file_url
+                    )
+                else:
+                    raise ProviderException(
+                        "Either file or file_url must be provided", code=400
+                    )
+                response = await poller.result()
+        except AzureError as provider_call_exception:
+            raise ProviderException(str(provider_call_exception))
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)